Fixed Multi-AZ Test on PowerMax for OCP

OliverChen557 opened the pull request:
# Description
- Fixed Multi-AZ Test on PowerMax for OCP where the zone and region label keys are different between k8s and OCP.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1748 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility
- [x] I have executed the relevant end-to-end test scenarios

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Successful execution of test scenario `Install Powermax Driver (Standalone) With Node Zoning` in the pipeline
```bash
[ECHO AND SEND] cd ~/csm-operator/tests/e2e; ./run-e2e-test.sh --pmax --no-modules --zoning
/root/csm-operator/tests/e2e
2025/04/29 08:44:23 maxprocs: Leaving GOMAXPROCS=4: CPU quota undefined
  W0429 08:44:57.004121  107577 test_context.go:541] Unable to find in-cluster config, using default host : https://127.0.0.1:6443
  I0429 08:44:57.004284 107577 test_context.go:564] The --provider flag is not set. Continuing as if --provider=skeleton had been used.
Running Suite: CSM Operator End-to-End Tests - /root/csm-operator/tests/e2e
===========================================================================
Random Seed: 1745934263

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/csm-operator/tests/e2e/e2e_test.go:98
  STEP: Getting test environment variables @ 04/29/25 08:44:57.005
  STEP: [nomodules powermax zoning] @ 04/29/25 08:44:57.005
  STEP: Reading values file @ 04/29/25 08:44:57.005
  STEP: Getting a k8s client @ 04/29/25 08:44:57.015
[BeforeSuite] PASSED [0.018 seconds]
------------------------------
[run-e2e-test] E2E Testing Running all test Given Test Scenarios
/root/csm-operator/tests/e2e/e2e_test.go:139
  STEP: Starting: Install Powermax Driver (Standalone) With Node Zoning  @ 04/29/25 08:44:57.024
  STEP: Returning false at end @ 04/29/25 08:44:57.024
  STEP:      Executing  Given an environment with k8s or openshift, and CSM operator installed @ 04/29/25 08:44:57.024
  STEP:      Executing  Run [Apply Node Labels] @ 04/29/25 08:44:57.043
  I0429 08:44:57.043247 107577 util.go:601] Running kubectl [get nodes --show-labels]
NAME                                                  STATUS   ROLES                  AGE     VERSION   LABELS
ocppmx-master-xxx1.ocppmx.hop.lab.emc.com   Ready    control-plane,master   4h37m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx1.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx2.ocppmx.hop.lab.emc.com   Ready    control-plane,master   4h36m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx2.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx3.ocppmx.hop.lab.emc.com   Ready    control-plane,master   4h35m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx3.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com   Ready    worker                 4h16m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos,zone.topology.kubernetes.io/region=zoneC,zone.topology.kubernetes.io/zone=zoneA
ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com   Ready    worker                 4h16m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos,zone.topology.kubernetes.io/region=zoneD,zone.topology.kubernetes.io/zone=zoneB
  I0429 08:44:57.651427 107577 util.go:601] Running /bin/bash [modify_zoning_labels.sh add zone.topology.kubernetes.io/zone=zoneA zone.topology.kubernetes.io/zone=zoneB]
node/ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com not labeled
Added zone label 'zone.topology.kubernetes.io/zone=zoneA' to ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com not labeled
Added zone label 'zone.topology.kubernetes.io/zone=zoneB' to ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
  I0429 08:44:58.420519 107577 util.go:601] Running /bin/bash [modify_zoning_labels.sh add zone.topology.kubernetes.io/region=zoneC zone.topology.kubernetes.io/region=zoneD]
node/ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com not labeled
Added zone label 'zone.topology.kubernetes.io/region=zoneC' to ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com not labeled
Added zone label 'zone.topology.kubernetes.io/region=zoneD' to ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
  I0429 08:44:59.149883 107577 util.go:601] Running kubectl [get nodes --show-labels]
NAME                                                  STATUS   ROLES                  AGE     VERSION   LABELS
ocppmx-master-xxx1.ocppmx.hop.lab.emc.com   Ready    control-plane,master   4h37m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx1.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx2.ocppmx.hop.lab.emc.com   Ready    control-plane,master   4h36m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx2.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx3.ocppmx.hop.lab.emc.com   Ready    control-plane,master   4h35m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx3.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com   Ready    worker                 4h16m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos,zone.topology.kubernetes.io/region=zoneC,zone.topology.kubernetes.io/zone=zoneA
ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com   Ready    worker                 4h16m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos,zone.topology.kubernetes.io/region=zoneD,zone.topology.kubernetes.io/zone=zoneB
  STEP:      Executing  Set up reverse proxy tls secret namespace [powermax] @ 04/29/25 08:44:59.348
Temporary directory created at: /root/csm-operator/tests/e2e/tls-setup3089321658
  STEP:      Executing  Set up secret with template [testfiles/powermax-templates/powermax-zoning-use-secret-template.yaml] name [powermax-config] in namespace [powermax] for [pmaxUseSecret] @ 04/29/25 08:45:00.33
  STEP:      Executing  Set up creds with template [testfiles/powermax-templates/powermax-array-config.yaml] for [pmaxArrayConfig] @ 04/29/25 08:45:00.786
  STEP:      Executing  Apply custom resource [1] @ 04/29/25 08:45:01.138
  I0429 08:45:01.138985 107577 builder.go:121] Running '/usr/bin/kubectl --namespace=powermax apply --validate=true -f -'
  I0429 08:45:01.516500 107577 builder.go:146] stderr: ""
  I0429 08:45:01.516572 107577 builder.go:147] stdout: "containerstoragemodule.storage.dell.com/powermax created\n"
  STEP:      Executing  Validate custom resource [1] @ 04/29/25 08:45:01.516

err: expected custom resource status to be Succeeded. Got: Failed
  STEP:      Executing  Validate [powermax] driver from CR [1] is installed @ 04/29/25 08:45:51.583
  STEP:      Executing  Run [Validate Node Labels] @ 04/29/25 08:46:11.589
  I0429 08:46:11.589885 107577 util.go:601] Running /bin/bash [modify_zoning_labels.sh validate-zoning powermax]
Using label keys: zone.topology.kubernetes.io/zone and zone.topology.kubernetes.io/region
Configured zones in secret: zoneA zoneC zoneB zoneD
Checking node: ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com zone label: zoneB
Comparing node zone label 'zoneB' with expected zone 'zoneA'
Comparing node zone label 'zoneB' with expected zone 'zoneC'
Comparing node zone label 'zoneB' with expected zone 'zoneB'
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com has a matching zone label: zoneB
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com zone label: zoneD
Comparing node zone label 'zoneD' with expected zone 'zoneA'
Comparing node zone label 'zoneD' with expected zone 'zoneC'
Comparing node zone label 'zoneD' with expected zone 'zoneB'
Comparing node zone label 'zoneD' with expected zone 'zoneD'
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com has a matching zone label: zoneD
Checking node: ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneA
Comparing node zone label 'zoneA' with expected zone 'zoneA'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneA
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneC
Comparing node zone label 'zoneC' with expected zone 'zoneA'
Comparing node zone label 'zoneC' with expected zone 'zoneC'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneC
Checking node: ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneA
Comparing node zone label 'zoneA' with expected zone 'zoneA'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneA
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneC
Comparing node zone label 'zoneC' with expected zone 'zoneA'
Comparing node zone label 'zoneC' with expected zone 'zoneC'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneC
Checking node: ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com zone label: zoneB
Comparing node zone label 'zoneB' with expected zone 'zoneA'
Comparing node zone label 'zoneB' with expected zone 'zoneC'
Comparing node zone label 'zoneB' with expected zone 'zoneB'
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com has a matching zone label: zoneB
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com zone label: zoneD
Comparing node zone label 'zoneD' with expected zone 'zoneA'
Comparing node zone label 'zoneD' with expected zone 'zoneC'
Comparing node zone label 'zoneD' with expected zone 'zoneB'
Comparing node zone label 'zoneD' with expected zone 'zoneD'
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com has a matching zone label: zoneD
Checking node: ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneA
Comparing node zone label 'zoneA' with expected zone 'zoneA'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneA
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneC
Comparing node zone label 'zoneC' with expected zone 'zoneA'
Comparing node zone label 'zoneC' with expected zone 'zoneC'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneC
All zones on worker nodes match expected values from secret.
  STEP:      Executing  Enable forceRemoveDriver on CR [1] @ 04/29/25 08:46:16.228
  STEP:      Executing  Delete custom resource [1] @ 04/29/25 08:46:16.255
  STEP:      Executing  Validate [powermax] driver from CR [1] is not installed @ 04/29/25 08:46:16.281

err: found the following pods: csipowermax-reverseproxy-55c9b59955-nf892,
  STEP:      Executing  Run [Remove Node Labels] @ 04/29/25 08:47:06.309
  I0429 08:47:06.309365 107577 util.go:601] Running /bin/bash [modify_zoning_labels.sh remove-all-zones]
node/ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/region' from ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/zone' from ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/region' from ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/zone' from ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
  I0429 08:47:07.771489 107577 util.go:601] Running kubectl [get nodes --show-labels]
NAME                                                  STATUS   ROLES                  AGE     VERSION   LABELS
ocppmx-master-xxx1.ocppmx.hop.lab.emc.com   Ready    control-plane,master   4h39m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx1.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx2.ocppmx.hop.lab.emc.com   Ready    control-plane,master   4h38m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx2.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx3.ocppmx.hop.lab.emc.com   Ready    control-plane,master   4h38m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx3.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com   Ready    worker                 4h18m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos
ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com   Ready    worker                 4h18m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001309.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001309.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001309=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos
  STEP: Ending: Install Powermax Driver (Standalone) With Node Zoning
   @ 04/29/25 08:47:07.984
• [135.965 seconds]
------------------------------

Ran 1 of 1 Specs in 135.984 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 2m49.736847949s
Test Suite Passed
```
- [x] Successful execution of test scenario `Install PowerFlex Driver (Standalone) With Node Zoning` in the pipeline
```bash
[ECHO AND SEND] cd ~/csm-operator/tests/e2e; ./run-e2e-test.sh --pflex --no-modules
/root/csm-operator/tests/e2e
2025/04/29 09:30:30 maxprocs: Leaving GOMAXPROCS=4: CPU quota undefined
  W0429 09:32:12.932759  415618 test_context.go:541] Unable to find in-cluster config, using default host : https://127.0.0.1:6443
  I0429 09:32:12.932953 415618 test_context.go:564] The --provider flag is not set. Continuing as if --provider=skeleton had been used.
Running Suite: CSM Operator End-to-End Tests - /root/csm-operator/tests/e2e
===========================================================================
Random Seed: 1745937030

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/csm-operator/tests/e2e/e2e_test.go:98
  STEP: Getting test environment variables @ 04/29/25 09:32:12.933
  STEP: [nomodules powerflex] @ 04/29/25 09:32:12.933
  STEP: Reading values file @ 04/29/25 09:32:12.933
  STEP: Getting a k8s client @ 04/29/25 09:32:12.944
[BeforeSuite] PASSED [0.016 seconds]
------------------------------
[run-e2e-test] E2E Testing Running all test Given Test Scenarios
/root/csm-operator/tests/e2e/e2e_test.go:139
  STEP: Starting: Install PowerFlex Driver (Standalone) With Node Zoning  @ 04/29/25 09:32:12.949
  STEP: Returning false at end @ 04/29/25 09:32:12.949
  STEP:      Executing  Given an environment with k8s or openshift, and CSM operator installed @ 04/29/25 09:32:12.949
  STEP:      Executing  Run [Apply Node Labels] @ 04/29/25 09:32:12.966
  I0429 09:32:12.966461 415618 util.go:601] Running kubectl [get nodes --show-labels]
NAME                                                STATUS   ROLES                  AGE     VERSION   LABELS
ocppfx-master-xxx1.ocppfx.hop.lab.emc.com   Ready    control-plane,master   3h26m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-master-xxx1.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppfx-master-xxx2.ocppfx.hop.lab.emc.com   Ready    control-plane,master   3h26m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-master-xxx2.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppfx-master-xxx3.ocppfx.hop.lab.emc.com   Ready    control-plane,master   3h25m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-master-xxx3.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com   Ready    worker                 3h9m    v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-isilon.dellemc.com/10.247.27.96=csi-isilon.dellemc.com,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f-nfs=true,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos
ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com   Ready    worker                 3h9m    v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-isilon.dellemc.com/10.247.27.96=csi-isilon.dellemc.com,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f-nfs=true,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos
  I0429 09:32:13.752787 415618 util.go:601] Running /bin/bash [modify_zoning_labels.sh add zone.csi-vxflexos.dellemc.com=zoneA zone.csi-vxflexos.dellemc.com=zoneB]
node/ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com labeled
Added zone label 'zone.csi-vxflexos.dellemc.com=zoneA' to ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com
node/ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com labeled
Added zone label 'zone.csi-vxflexos.dellemc.com=zoneB' to ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com
  I0429 09:32:14.466061 415618 util.go:601] Running kubectl [get nodes --show-labels]
NAME                                                STATUS   ROLES                  AGE     VERSION   LABELS
ocppfx-master-xxx1.ocppfx.hop.lab.emc.com   Ready    control-plane,master   3h26m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-master-xxx1.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppfx-master-xxx2.ocppfx.hop.lab.emc.com   Ready    control-plane,master   3h26m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-master-xxx2.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppfx-master-xxx3.ocppfx.hop.lab.emc.com   Ready    control-plane,master   3h25m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-master-xxx3.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com   Ready    worker                 3h9m    v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-isilon.dellemc.com/10.247.27.96=csi-isilon.dellemc.com,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f-nfs=true,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos,zone.csi-vxflexos.dellemc.com=zoneA
ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com   Ready    worker                 3h9m    v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-isilon.dellemc.com/10.247.27.96=csi-isilon.dellemc.com,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f-nfs=true,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos,zone.csi-vxflexos.dellemc.com=zoneB
  STEP:      Executing  Set up temporary secret with template [testfiles/powerflex-templates/powerflex-zoning-secret-template.yaml] name [test-vxflexos-config] in namespace [test-vxflexos] for [pflex] @ 04/29/25 09:32:14.693
  STEP:      Executing  Apply custom resource [1] @ 04/29/25 09:32:15.319
  I0429 09:32:15.320334 415618 builder.go:121] Running '/usr/bin/kubectl --namespace=test-vxflexos apply --validate=true -f -'
  I0429 09:32:15.624490 415618 builder.go:146] stderr: ""
  I0429 09:32:15.624578 415618 builder.go:147] stdout: "containerstoragemodule.storage.dell.com/test-vxflexos created\n"
  STEP:      Executing  Validate [powerflex] driver from CR [1] is installed @ 04/29/25 09:32:15.624

err:
The pod(test-vxflexos-node-4wczw) is Pending
The pod(test-vxflexos-node-srlzn) is Pending
  STEP:      Executing  Run [Validate Node Labels] @ 04/29/25 09:33:05.675
  I0429 09:33:05.675131 415618 util.go:601] Running /bin/bash [modify_zoning_labels.sh validate-zoning powerflex]
    name: "zoneA"
    name: "zoneB"
Configured zones in secret: zoneA zoneB
Checking node: ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com
Node ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com zone label: zoneB
Comparing node zone label 'zoneB' with expected zone 'zoneA'
Comparing node zone label 'zoneB' with expected zone 'zoneB'
Node ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com has a matching zone label: zoneB
Checking node: ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com
Node ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com zone label: zoneA
Comparing node zone label 'zoneA' with expected zone 'zoneA'
Node ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com has a matching zone label: zoneA
Checking node: ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com
Node ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com zone label: zoneA
Comparing node zone label 'zoneA' with expected zone 'zoneA'
Node ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com has a matching zone label: zoneA
Checking node: ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com
Node ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com zone label: zoneB
Comparing node zone label 'zoneB' with expected zone 'zoneA'
Comparing node zone label 'zoneB' with expected zone 'zoneB'
Node ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com has a matching zone label: zoneB
  STEP:      Executing  Enable forceRemoveDriver on CR [1] @ 04/29/25 09:33:08.081
  STEP:      Executing  Delete custom resource [1] @ 04/29/25 09:33:08.119
  STEP:      Executing  Validate [powerflex] driver from CR [1] is not installed @ 04/29/25 09:33:08.144
  STEP:      Executing  Run [Remove Node Labels] @ 04/29/25 09:33:28.163
  I0429 09:33:28.163641 415618 util.go:601] Running /bin/bash [modify_zoning_labels.sh remove-all-zones]
node/ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com unlabeled
Removed label 'zone.csi-vxflexos.dellemc.com' from ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com
node/ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com unlabeled
Removed label 'zone.csi-vxflexos.dellemc.com' from ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com
  I0429 09:33:29.130225 415618 util.go:601] Running kubectl [get nodes --show-labels]
NAME                                                STATUS   ROLES                  AGE     VERSION   LABELS
ocppfx-master-xxx1.ocppfx.hop.lab.emc.com   Ready    control-plane,master   3h27m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-master-xxx1.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppfx-master-xxx2.ocppfx.hop.lab.emc.com   Ready    control-plane,master   3h27m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-master-xxx2.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppfx-master-xxx3.ocppfx.hop.lab.emc.com   Ready    control-plane,master   3h26m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-master-xxx3.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com   Ready    worker                 3h10m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-isilon.dellemc.com/10.247.27.96=csi-isilon.dellemc.com,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f-nfs=true,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-worker-xxx4.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos
ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com   Ready    worker                 3h10m   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-isilon.dellemc.com/10.247.27.96=csi-isilon.dellemc.com,csi-vxflexos.dellemc.com/6689fba68522eb0f-nfs=true,csi-vxflexos.dellemc.com/6689fba68522eb0f=csi-vxflexos.dellemc.com,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f-nfs=true,csi-vxflexos.dellemc.com/ffc2fbeef33b6b0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppfx-worker-xxx5.ocppfx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos
  STEP: Ending: Install PowerFlex Driver (Standalone) With Node Zoning
   @ 04/29/25 09:33:29.324
  STEP: Starting: Install Powermax Driver (Standalone) With Node Zoning  @ 04/29/25 09:33:34.325
  STEP: No matching tags for scenario @ 04/29/25 09:33:34.325
  STEP: Not tagged for this test run, skipping @ 04/29/25 09:33:34.325
  STEP: Ending: Install Powermax Driver (Standalone) With Node Zoning
   @ 04/29/25 09:33:34.325
• [81.376 seconds]
------------------------------

Ran 1 of 1 Specs in 81.392 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 3m3.954680135s
```
- [x] Manual testing of getting the correct label keys on a k8s cluster:
```
e2e $ kubectl get nodes --show-labels
NAME                     STATUS   ROLES           AGE    VERSION   LABELS
master-1-ksz3qb8bb4khq   Ready    control-plane   285d   v1.30.0   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=master-1-ksz3qb8bb4khq,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node.kubernetes.io/exclude-from-external-load-balancers=
worker-1-ksz3qb8bb4khq   Ready    <none>          285d   v1.30.0   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-vxflexos.dellemc.com/1235e15806d1ec0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=worker-1-ksz3qb8bb4khq,kubernetes.io/os=linux,zone.topology.kubernetes.io/region=zoneC,zone.topology.kubernetes.io/zone=zoneA
worker-2-ksz3qb8bb4khq   Ready    <none>          285d   v1.30.0   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-vxflexos.dellemc.com/1235e15806d1ec0f=csi-vxflexos.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=worker-2-ksz3qb8bb4khq,kubernetes.io/os=linux,zone.topology.kubernetes.io/region=zoneD,zone.topology.kubernetes.io/zone=zoneB
e2e $ /bin/bash modify_zoning_labels.sh validate-zoning powermax
Error from server (NotFound): namespaces "powermax" not found
Using label keys: topology.kubernetes.io/zone and topology.kubernetes.io/region
```

ChristianAtDell commented on tests/e2e/modify_zoning_labels.sh:
```diff
@@ -102,13 +102,27 @@ validate_zoning_powermax() {
+    zone_label_key="zone.topology.kubernetes.io/zone"
+    region_label_key="zone.topology.kubernetes.io/region"
+  else
+    zone_label_key="topology.kubernetes.io/zone"
+    region_label_key="topology.kubernetes.io/region"
```
This will not work; removing labels does so specifically based on the 'zone' prefix because label keys are arbitrary.

OliverChen557 replied:
Could you elaborate on what won't work? The current implementation doesn't work on OCP because no labels were obtained.
Before my changes:
```
Configured zones in secret: zoneA zoneC zoneB zoneD
Checking node: ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: null
Comparing node zone label 'null' with expected zone 'zoneA'
Comparing node zone label 'null' with expected zone 'zoneC'
Comparing node zone label 'null' with expected zone 'zoneB'
Comparing node zone label 'null' with expected zone 'zoneD'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com does not have a matching zone label: null
```
After my changes:
```
Checking node: ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneA
Comparing node zone label 'zoneA' with expected zone 'zoneA'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneA
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneC
Comparing node zone label 'zoneC' with expected zone 'zoneA'
Comparing node zone label 'zoneC' with expected zone 'zoneC'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneC
```

ChristianAtDell replied:
https://github.com/dell/csm-operator/pull/968/files#diff-54182f8f647592ff737ead0d900f5a84329ef829615a586c63358f53205ac890L77

With the labels as have been specified, not including "zone" as a prefix leads to a problem with remove_all_zones
https://github.com/dell/csm-operator/blob/pvt/upgrade-test-version/tests/e2e/testfiles/scenarios.yaml#L2918
https://github.com/dell/csm-operator/blob/pvt/upgrade-test-version/tests/e2e/testfiles/scenarios.yaml#L2882

So the test will leave zones on the nodes, right?

Either something has to change in the scenarios so that it more intelligently removes zoning labels or all zones must start with prefix "zone"

OliverChen557 replied:
I didn't change how we are adding on the zone labels or removing them. I am only specifying the labels to look for when validating the labels. Could you help me understand why my changes would affect `remove_all_zone_labels` where it's getting the labels on its own? Also, I didn't change the existing behavior introduced in https://github.com/dell/csm-operator/pull/958, which looks for `topology.kubernetes.io/zone` [example](https://github.com/dell/csm-operator/blob/fbf3ba96c5068f7a47bcc056d199f78b86fb162c/tests/e2e/modify_zoning_labels.sh#L107) and `topology.kubernetes.io/region` [example](https://github.com/dell/csm-operator/blob/fbf3ba96c5068f7a47bcc056d199f78b86fb162c/tests/e2e/modify_zoning_labels.sh#L111C22-L111C51).
And the labels were successfully removed from my logs above.
```bash
  STEP:      Executing  Run [Remove Node Labels] @ 04/29/25 08:47:06.309
  I0429 08:47:06.309365 107577 util.go:601] Running /bin/bash [modify_zoning_labels.sh remove-all-zones]
node/ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/region' from ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/zone' from ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/region' from ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/zone' from ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
```

OliverChen557 replied:
Discussed with @ChristianAtDell, the conclusion is that the zone prefix addition to the test code was missed in a previous PR. It is NOT an issue of k8s vs. OCP. Update the descriptions.

ChristianAtDell reviewed (changes requested):
https://github.com/dell/csm-operator/pull/968/files#diff-54182f8f647592ff737ead0d900f5a84329ef829615a586c63358f53205ac890L77 See the above comment.

OliverChen557 commented:
After the changes, another successful execution of test scenario `Install Powermax Driver (Standalone) With Node Zoning` in the pipeline:
```bash
[ECHO AND SEND] cd ~/csm-operator/tests/e2e; ./run-e2e-test.sh --pmax --no-modules
/root/csm-operator/tests/e2e
2025/04/29 14:10:36 maxprocs: Leaving GOMAXPROCS=4: CPU quota undefined
  W0429 14:10:58.521683  126041 test_context.go:541] Unable to find in-cluster config, using default host : https://127.0.0.1:6443
  I0429 14:10:58.521851 126041 test_context.go:564] The --provider flag is not set. Continuing as if --provider=skeleton had been used.
Running Suite: CSM Operator End-to-End Tests - /root/csm-operator/tests/e2e
===========================================================================
Random Seed: 1745953836

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/csm-operator/tests/e2e/e2e_test.go:98
  STEP: Getting test environment variables @ 04/29/25 14:10:58.524
  STEP: [nomodules powermax] @ 04/29/25 14:10:58.524
  STEP: Reading values file @ 04/29/25 14:10:58.524
  STEP: Getting a k8s client @ 04/29/25 14:10:58.532
[BeforeSuite] PASSED [0.016 seconds]
------------------------------
[run-e2e-test] E2E Testing Running all test Given Test Scenarios
/root/csm-operator/tests/e2e/e2e_test.go:139
  STEP: Starting: Install Powermax Driver (Standalone) With Node Zoning  @ 04/29/25 14:10:58.54
  STEP: Returning false at end @ 04/29/25 14:10:58.54
  STEP:      Executing  Given an environment with k8s or openshift, and CSM operator installed @ 04/29/25 14:10:58.54
  STEP:      Executing  Run [Apply Node Labels] @ 04/29/25 14:10:58.563
  I0429 14:10:58.563274 126041 util.go:601] Running kubectl [get nodes --show-labels]
NAME                                                  STATUS   ROLES                  AGE   VERSION   LABELS
ocppmx-master-xxx1.ocppmx.hop.lab.emc.com   Ready    control-plane,master   10h   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx1.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx2.ocppmx.hop.lab.emc.com   Ready    control-plane,master   10h   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx2.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx3.ocppmx.hop.lab.emc.com   Ready    control-plane,master   10h   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx3.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com   Ready    worker                 9h    v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos
ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com   Ready    worker                 9h    v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001309.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001309.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001309=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos
  I0429 14:10:58.989363 126041 util.go:601] Running /bin/bash [modify_zoning_labels.sh add zone.topology.kubernetes.io/zone=zoneA zone.topology.kubernetes.io/zone=zoneB]
node/ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com labeled
Added zone label 'zone.topology.kubernetes.io/zone=zoneA' to ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com labeled
Added zone label 'zone.topology.kubernetes.io/zone=zoneB' to ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
  I0429 14:10:59.792979 126041 util.go:601] Running /bin/bash [modify_zoning_labels.sh add zone.topology.kubernetes.io/region=zoneC zone.topology.kubernetes.io/region=zoneD]
node/ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com labeled
Added zone label 'zone.topology.kubernetes.io/region=zoneC' to ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com labeled
Added zone label 'zone.topology.kubernetes.io/region=zoneD' to ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
  I0429 14:11:00.522554 126041 util.go:601] Running kubectl [get nodes --show-labels]
NAME                                                  STATUS   ROLES                  AGE   VERSION   LABELS
ocppmx-master-xxx1.ocppmx.hop.lab.emc.com   Ready    control-plane,master   10h   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx1.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx2.ocppmx.hop.lab.emc.com   Ready    control-plane,master   10h   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx2.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx3.ocppmx.hop.lab.emc.com   Ready    control-plane,master   10h   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx3.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com   Ready    worker                 9h    v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos,zone.topology.kubernetes.io/region=zoneC,zone.topology.kubernetes.io/zone=zoneA
ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com   Ready    worker                 9h    v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001309.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001309.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001309=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos,zone.topology.kubernetes.io/region=zoneD,zone.topology.kubernetes.io/zone=zoneB
  STEP:      Executing  Set up reverse proxy tls secret namespace [powermax] @ 04/29/25 14:11:00.731
revproxy-certs secret already exists, skipping creation.
csirevproxy-tls-secret already exists, skipping creation.
  STEP:      Executing  Set up secret with template [testfiles/powermax-templates/powermax-zoning-use-secret-template.yaml] name [powermax-config] in namespace [powermax] for [pmaxUseSecret] @ 04/29/25 14:11:01.158
  STEP:      Executing  Set up creds with template [testfiles/powermax-templates/powermax-array-config.yaml] for [pmaxArrayConfig] @ 04/29/25 14:11:01.846
  STEP:      Executing  Apply custom resource [1] @ 04/29/25 14:11:02.371
  I0429 14:11:02.371348 126041 builder.go:121] Running '/usr/bin/kubectl --namespace=powermax apply --validate=true -f -'
  I0429 14:11:02.605833 126041 builder.go:146] stderr: ""
  I0429 14:11:02.605931 126041 builder.go:147] stdout: "containerstoragemodule.storage.dell.com/powermax created\n"
  STEP:      Executing  Validate custom resource [1] @ 04/29/25 14:11:02.606

err: expected custom resource status to be Succeeded. Got: Failed
  STEP:      Executing  Validate [powermax] driver from CR [1] is installed @ 04/29/25 14:11:52.681
  STEP:      Executing  Run [Validate Node Labels] @ 04/29/25 14:12:12.703
  I0429 14:12:12.703774 126041 util.go:601] Running /bin/bash [modify_zoning_labels.sh validate-zoning powermax]
Using label keys: zone.topology.kubernetes.io/zone and zone.topology.kubernetes.io/region
Configured zones in secret: zoneA zoneC zoneB zoneD
Checking node: ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com zone label: zoneB
Comparing node zone label 'zoneB' with expected zone 'zoneA'
Comparing node zone label 'zoneB' with expected zone 'zoneC'
Comparing node zone label 'zoneB' with expected zone 'zoneB'
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com has a matching zone label: zoneB
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com zone label: zoneD
Comparing node zone label 'zoneD' with expected zone 'zoneA'
Comparing node zone label 'zoneD' with expected zone 'zoneC'
Comparing node zone label 'zoneD' with expected zone 'zoneB'
Comparing node zone label 'zoneD' with expected zone 'zoneD'
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com has a matching zone label: zoneD
Checking node: ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneA
Comparing node zone label 'zoneA' with expected zone 'zoneA'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneA
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneC
Comparing node zone label 'zoneC' with expected zone 'zoneA'
Comparing node zone label 'zoneC' with expected zone 'zoneC'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneC
Checking node: ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneA
Comparing node zone label 'zoneA' with expected zone 'zoneA'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneA
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneC
Comparing node zone label 'zoneC' with expected zone 'zoneA'
Comparing node zone label 'zoneC' with expected zone 'zoneC'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneC
Checking node: ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneA
Comparing node zone label 'zoneA' with expected zone 'zoneA'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneA
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com zone label: zoneC
Comparing node zone label 'zoneC' with expected zone 'zoneA'
Comparing node zone label 'zoneC' with expected zone 'zoneC'
Node ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com has a matching zone label: zoneC
Checking node: ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com zone label: zoneB
Comparing node zone label 'zoneB' with expected zone 'zoneA'
Comparing node zone label 'zoneB' with expected zone 'zoneC'
Comparing node zone label 'zoneB' with expected zone 'zoneB'
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com has a matching zone label: zoneB
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com zone label: zoneD
Comparing node zone label 'zoneD' with expected zone 'zoneA'
Comparing node zone label 'zoneD' with expected zone 'zoneC'
Comparing node zone label 'zoneD' with expected zone 'zoneB'
Comparing node zone label 'zoneD' with expected zone 'zoneD'
Node ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com has a matching zone label: zoneD
All zones on worker nodes match expected values from secret.
  STEP:      Executing  Enable forceRemoveDriver on CR [1] @ 04/29/25 14:12:17.106
  STEP:      Executing  Delete custom resource [1] @ 04/29/25 14:12:17.131
  STEP:      Executing  Validate [powermax] driver from CR [1] is not installed @ 04/29/25 14:12:17.159

err: found the following pods: csipowermax-reverseproxy-55c9b59955-mnqsx,
  STEP:      Executing  Run [Remove Node Labels] @ 04/29/25 14:13:07.185
  I0429 14:13:07.185755 126041 util.go:601] Running /bin/bash [modify_zoning_labels.sh remove-all-zones]
node/ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/region' from ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/zone' from ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/region' from ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
node/ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com unlabeled
Removed label 'zone.topology.kubernetes.io/zone' from ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com
  I0429 14:13:08.584148 126041 util.go:601] Running kubectl [get nodes --show-labels]
NAME                                                  STATUS   ROLES                  AGE   VERSION   LABELS
ocppmx-master-xxx1.ocppmx.hop.lab.emc.com   Ready    control-plane,master   10h   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx1.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx2.ocppmx.hop.lab.emc.com   Ready    control-plane,master   10h   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx2.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-master-xxx3.ocppmx.hop.lab.emc.com   Ready    control-plane,master   10h   v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-master-xxx3.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.openshift.io/os_id=rhcos
ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com   Ready    worker                 9h    v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx4.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos
ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com   Ready    worker                 9h    v1.31.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,csi-powermax.dellemc.com/000120001309.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001309.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001309=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.iscsi=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836.nfs=csi-powermax.dellemc.com,csi-powermax.dellemc.com/000120001836=csi-powermax.dellemc.com,kubernetes.io/arch=amd64,kubernetes.io/hostname=ocppmx-worker-xxx5.ocppmx.hop.lab.emc.com,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.openshift.io/os_id=rhcos
  STEP: Ending: Install Powermax Driver (Standalone) With Node Zoning
   @ 04/29/25 14:13:08.773
• [135.237 seconds]
------------------------------

Ran 1 of 1 Specs in 135.255 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 2m37.08981322s
Test Suite Passed
```

alikdell commented on tests/e2e/modify_zoning_labels.sh:
```diff
@@ -102,13 +102,19 @@ validate_zoning_powermax() {
+  zone_label_key="zone.topology.kubernetes.io/zone"
+  region_label_key="zone.topology.kubernetes.io/region"
+
+  echo "Using label keys: $zone_label_key and $region_label_key"
```
Isn't this print will always print above 2 lines? Not a blocker just saying

OliverChen557 replied:
Yes, I wanted to make sure we are printing this so we know what keys we are using after setting them.
